src/operations.jl

```diff
@@ -56,29 +79,83 @@ Base.:(==)(x::ArrayReferenceMeta, y) = false
 
 abstract type AbstractLoopOperation end
 
+"""
+`OperationType` is an `@enum` for classifying supported operations that can appear in
+`@avx` blocks. Type `LoopVectorization.OperationType` to see the different types.
+"""
 @enum OperationType begin
     constant
     memload
     compute
     memstore
     loopvalue
 end
+"An operation setting a variable to a constant value (e.g., `a = 0.0`)" constant
+"An operation setting a variable from a memory location (e.g., `a = A[i,j]`)" memload
+"An operation computing a new value from one or more variables (e.g., `a = b + c`)" compute
+"An operation storing a value to a memory location (e.g., `A[i,j] = a`)" memstore
+"??" loopvalue
 
 # TODO: can some computations be cached in the operations?
 """
+    Operation
+
+A structure to encode a particular action occuring inside an `@avx` block.
+
+# Fields
+
+$(TYPEDFIELDS)
+
+# Example
+
+```jldoctest Operation; filter = r"\\"##.*\\""
+julia> using LoopVectorization
+
+julia> AmulBq = :(for m ∈ 1:M, n ∈ 1:N
+           C[m,n] = zero(eltype(B))
+           for k ∈ 1:K
+               C[m,n] += A[m,k] * B[k,n]
+           end
+       end);
+
+julia> lsAmulB = LoopVectorization.LoopSet(AmulBq);
+
+julia> LoopVectorization.operations(lsAmulB)
+6-element Array{LoopVectorization.Operation,1}:
+ var"##RHS#253" = var"##zero#254"
+ C[m, n] = var"##RHS#253"
+ var"##tempload#255" = A[m, k]
+ var"##tempload#256" = B[k, n]
+ var"##RHS#253" = LoopVectorization.vfmadd_fast(var"##tempload#255", var"##tempload#256", var"##RHS#253")
+ var"##RHS#253" = LoopVectorization.identity(var"##RHS#253")
+```
+Each one of these lines is a pretty-printed `Operation`.
 """
 mutable struct Operation <: AbstractLoopOperation
+    "A unique number assigned to each operation"
     identifier::Int
```

Comment:
Importantly, it corresponds to 1 less than its index in `operations(ls)`.
The function `identifier(op::Operation)` returns `op.identifier + 1`.

I liked the idea of defining the identifier with `length(operations(ls))`. I don't know where the best place for the `+1` is.
Inside the `identifier` function at least doesn't require repeating the code `+ 1`. But it'll get called frequently (not a big deal), and it should be clear that `identifier` should be accessed through the `identifier` function, rather than `getproperty`.